ERCS/erc-7007.md

```diff
@@ -133,14 +170,42 @@ ERC-7007 Metadata JSON Schema for reference
         "aigc_data": {
             "type": "string",
             "description": "A URI pointing to a resource with mime type image/* representing the asset to which this AIGC NFT represents."
+        },
+        "proof_type": {
+            "type": "string",
+            "description": "validity (zkML) or fraud (opML)"
         }
     }
 }
 ```
 
 ## Rationale
 
-TBD
+There are many proposed uses of non-fungible tokens (NFT) that depend on AI-generated content (AIGC). While traditional AIGC-based NFT can be easily implemented under the [ERC-721](./eip-721.md) schema, NFT holders don’t have the right to choose the prompt, which is the most creative part, given a pre-defined Machine Learning model. We want to give the right of prompt creation back to the NFT holders who don't have to afford the inference cost. Therefore we propose this verifiable AIGC-NFT interface to help separate prompt from the ML inference. The ecosystem will be stronger if we have a standardized interface that allows prompt-bound NFT.
+
+### Unique Token Identification
+
+This specification sets the `tokenId` to be the hash of its corresponding `prompt`, creating a deterministic and collision-resistant way to associate tokens with their unique content generation parameters. This design decision ensures that the same prompt (which corresponds to the same AI-generated content under the same model seed) cannot be minted more than once, thereby preventing duplication and preserving the uniqueness of each NFT within the ecosystem.
+
+### Generalization to Different Proof Types
+
+This specification accommodates two proof types: validity proofs for zkML and fraud proofs for opML. Function arguments in mint and verify are designed for generality, allowing for compatibility with both proof systems. Moreover, the specification includes an updatable extension that specifically serves the requirements of opML.
```

Comment:
Are mint and verify function names here? If so, please enclose them in backticks (\`).